### PR TITLE
meta-three: add example layer with custom machine

### PR DESCRIPTION
**Pull request by deribaucourt**
This layer is used to demonstrate common syntax related to machine and distro overrides.

### CODE REVIEW

**Comment by deribaucourt** [comment]
The recipes will definetly not build. I just made them up quickly to show limitations in the VSCode syntax highlighting
image

File: meta-three/conf/recipes-core/systemd/systemd-conf/examplemachine/wlan0.network [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[Match]
```
Comment (idillon-sfl) [inline meta-three/conf/recipes-core/systemd/systemd-conf/examplemachine/wlan0.network:1]: What behavior our extension should have toward this file and the places it is used?

File: meta-three/conf/distro/exampledistro.conf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+DISTROOVERRIDES =. "poky:"
+require conf/distro/poky.conf
+# fslc-base.inc reverses the order of the layers so we can't prepend
```
Comment (idillon-sfl) [inline meta-three/conf/distro/exampledistro.conf:3]: Is the file "fslc-base.inc" supposed to be included in this PR?

In any case, I suggest we use comments in this repo solely to describe the expected extension's behavior.

File: meta-three/conf/recipes-core/systemd/systemd-conf_%.bbappend [hunk 3]
```diff
@@ -0,0 +1,16 @@
+FILESEXTRAPATHS:prepend := "${THISDIR}/${PN}:"
```
Comment (idillon-sfl) [inline meta-three/conf/recipes-core/systemd/systemd-conf_%.bbappend:1]:
```suggestion
# "prepend" should be highlighted a different color than variables
# Hovering "prepend" should show its definition (https://docs.yoctoproject.org/bitbake/bitbake-user-manual/bitbake-user-manual-metadata.html?highlight=prepend#appending-and-prepending-override-style-syntax)
FILESEXTRAPATHS:prepend := "${THISDIR}/${PN}:"
```

File: meta-three/conf/recipes-core/systemd/systemd-conf_%.bbappend [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+SRC_URI:append:examplemachine = " \
+    ${@bb.utils.contains('MACHINE_FEATURES', 'wifi', 'file://wlan0.network', '', d)} \
+    https://raw.githubusercontent.com/RobertCNelson/netifd/master/systemd/network/wlan0.network;destsuffix=systemd/network \
+"
+
+do_install:append:examplemachine() {
```
Comment (idillon-sfl) [inline meta-three/conf/recipes-core/systemd/systemd-conf_%.bbappend:8]:
```suggestion
# "append" should be highlighted a different color than variables
# Hovering "append" should show its definition (https://docs.yoctoproject.org/bitbake/bitbake-user-manual/bitbake-user-manual-metadata.html?highlight=prepend#appending-and-prepending-override-style-syntax)
do_install:append:examplemachine() {
```

**Pull request closed without merging**